embedded/sql/row_reader.go

```diff
@@ -38,6 +40,12 @@ type RowReader interface {
 	onClose(func())
 }
 
+type ScanSpecs struct {
+	Index         *Index
```

Comment:
Why `Index` and `DescOrder` here is exported but `rangesByColID` is not?

Also previously the `ScanSpec` structure was in `stmt.go` and had all fields unexported.

Reply:
it's towards providing information on how the query was resolved (the chosen index and the order of the scan), ranges may be exposed in the future